.github/workflows/continuous-integration.yml

```diff
@@ -34,6 +34,27 @@ jobs:
           echo "Cleaning up previous run"
           rm -rf "${{ github.workspace }}"
 
+  doxygen:
+    name: Doxygen
+    # Only run Doxygen against the main branch
+    if: github.ref_name == 'main'
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: false
+    - uses: mattnotmitt/doxygen-action@v1
+      with:
+        working-directory: "./"
+        doxyfile-path: "./doc/Doxygen"
+    - name: Deploy Doxygen
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./html
```

Comment:
That's cool, thanks for working on it. A minor nitpick, should it be deployed under `./doc/html` ?

Reply:
Good call.  Thanks!